polkadot/node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -334,28 +368,52 @@ fn handle_child_process(
 	send_child_response(&mut pipe_write, response);
 }
 
+/// Returns clone stack size
+/// The stack size is represented by 2MiB * number of threads + native stack;
+fn get_max_stack_size(executor_params: &ExecutorParams, number_of_threads: u32) -> usize {
+	let deterministc_stack_limit = params_to_wasmtime_semantics(executor_params)
+		.deterministic_stack_limit
+		.expect("the default value is always available.");
```

Comment:
Proofs should follow a logical chain of reasoning and conclude with `qed`. Please look for examples in the repo, and let me know if it's still not clear. :)

Reply:
If the `deterministic_stack_limit` is always available, why does it need to be an optional value?

Reply:
I believe the `Semantics` type comes from the wasmtime executor crate, and I suppose there's some reason for `deterministic_stack_limit` to be optional in its API. There are other consumers of that crate, after all. However, here we know that it's always available, so we could either return it from `params_to_wasmtime_semantics`, or we could have a solid proof here. Hope that's clear!

Reply:
Yeah I believe we don't impose a stack limit on relay chain runtime execution, that's why it's optional.

Reply:
Proofs have to be locally reasonable, it should never break because some code somewhere else gets changed. Or put differently, whoever can (realistically) break your proof, must have seen the proof - because it is right next to it.

Reply:
Yes, and since there is already a proof in `params_to_wasmtime_semantics`, I would opt for returning the value from that function.

Reply:
Thanks!